dts/arm/atmel/sams70.dtsi

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright (c) 2022 RIC Electronics
+ * Copyright (c) 2017 Piotr Mienkowski
+ * Copyright (c) 2017 Justin Watson
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <dt-bindings/i2c/i2c.h>
+#include <dt-bindings/gpio/gpio.h>
+#include "dma_atmel_sams70.h"
+
+#include "sams70-pinctrl.dtsi"
+
+/ {
```

Comment:
Since SAM S70 is supposed to be a subset of the already supported SAM E70 series, the SAM S70 dtsi should be made a "base" dts, and the SAM E70 dtsi should be updated to include the SAM S70 dtsi for the sake of future maintainability.

We are doing the same thing for SAM V71, which is a superset of the SAM E70 series:
https://github.com/zephyrproject-rtos/zephyr/blob/284d2a11db131198a19f2b189e42fa0d5e925064/dts/arm/atmel/samv71.dtsi#L7

Note that the SAM V71 dtsi includes the SAM E70 dtsi instead of unnecessarily duplicating all the information.

Reply:
Agreed, I will change this. There is a similar situation with the pincontrol files that I will fix also.

Reply:
@stephanosio I have refactored the dts files to make it more maintainable. Please have a look when you have time. Once this is okay I will go ahead and create a board.

Reply:
Hi @mcscholtz , by @stephanosio suggestion, you don't need create a new board. You just need add another board target inside sam_e70_xplained directory. In fact, this board already have two targets `sam_e70_xplained` and `sam_e70b_xplained`.

Reply:
Thanks @nandojve and @stephanosio 

I have done as you suggested and added a separate target for the SAMS70 to the existing ```sam_e70_xplained``` board.

It behaves as expected and will compile ```zephyr/samples/basic/blinky``` but fails on the ```zephyr/samples/net/*``` since the S70 does not have ethernet.

Just a question about the dts files: 

I added all the peripherals supported by the S70 to the sams70.dtsi. Some peripherals are only supported depending on which package is used. 

For example:
spi1 is only available on Q packages but not on J or N packages. 

However spi1 is added to the sams70.dtsi file, so there should be a really be separate file for each package and the J one should be the base. 

I added comments to the pinctrl file to indicate which peripherals are only available to which packages. Is this something that can be changed in a different PR?

Also the build seems to be failing because the hal is missing

Reply:
> Also the build seems to be failing because the hal is missing

@mcscholtz Please update `west.yml` to point to the PR https://github.com/zephyrproject-rtos/hal_atmel/pull/18.

See https://docs.zephyrproject.org/latest/guides/modules.html#process-for-submitting-changes-to-existing-modules.